### DISCUSSION TITLE
[General] What consensus mechanism is used for zksync hyperchains?

### DISCUSSION THREAD

**Discussion by namn-grg** [zkSync-Community-Hub/zksync-developers#172] [ZK Stack] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

What consensus mechanism is used for zksync hyperchains?  and is consensus even needed in hyperchains

I read in the FAQs that zkPorter uses POS. Why is that?

I came across this question through topic "How to avoid the fragmentation of consensus stake in a multi-rollup ecosystem" in the zkfellowship. I dont even think consensus stake is even needed for L2's. L2's security is dependent on the L1 so why stake

### What is the specific issue or error you're encountering?

.

### Can you share the error messages or logs you're receiving, if any?

.

### Have you made any recent changes to the contract before encountering this issue?

.

### Are there any external libraries or contracts that your contract interacts with?

.

### Can you provide the relevant portions of your contract code where the issue is occurring?

.

### Have you tried to isolate the problem, and if so, what were the results?

.

### What steps have you already taken to try to resolve the issue?

.

**Comment by dutterbutter** [comment, marked as answer]
hello 👋 

Currently zkSync does not have a consensus mechanism in place but we intend to decentralize the sequencers where we will make use of a faster version of HotStuff that we call HotShot. Alex briefly discusses it in his ETHCC address which can be found [here](https://www.youtube.com/watch?v=3eLdhYyuK4E), roughly at the ~8:40 mark.

**Reply by namn-grg** [reply]
Hi @dutterbutter  , thanks for sharing this fantastic talk!

What I concluded from this is that hyperchains will remain **centralized** ?

but a new type of **chain**(?) - zkPorter will be launched with decentralized sequencers - **which consensus mechanism?** 

I didn't fully get it, please help me with the above questions. Thanks.

**Reply by dutterbutter** [reply]
No zkSync, and hyperchains will follow a progressive decentralization. This will happen over time starting with sequencers. zkPorter is an option for hyperchains that will allow for off-chain data availability. Its not a type of chain, rather an option to an application specific zkrollup (hyperchain). This will entirely be up to the hyperchain development team if they want data availability to take place on or off chain. 

The consensus mechanism used will still be a faster version of HotStuff that we call HotShot.

**Reply by dutterbutter** [reply]
Its important to note, that regardless of the design choice (e.g. zkPorter / zkRollup) contracts and accounts on the zkRollup side will be able to seamlessly interact with accounts on the zkPorter side, and vice versa.

**Reply by dutterbutter** [reply]
There will be multiple _flavours_ of hyperchains as teams determine what customizations, or application specific changes, are required to meet their needs.

**Reply by namn-grg** [reply]
Got it. This definitely solved a lot of confusion.

So is HotShot/HotStuff a type of PoS? [Here](https://docs.zksync.io/zkevm/#what-is-zkporter) its mentioned that zkPorter will work on PoS. In this its also mentioned the zksync tokens will be used to stake - so will the stake needed to be done individually to all zkPorter chains?

Also another ques - Will the security of L1 determine hyperchains security? (I think the ans is - obviously yes)

Thanks!

**Reply by dutterbutter** [reply]
I don't know too much details about our implementation, but I would assume its a type of PoS given its based off of HotStuff. 

Security will be inherited by L1 (Ethereum), yes.

**Reply by namn-grg** [reply]
Ok got it!

Last thing - I had a doubt regarding a topic given in [zksyncfellowship](https://zksync.mirror.xyz/fN7WUNVsl6Vt8nhNtOtehtDoCEwZKMEgO5d92_QTEnA). Can I post it here or should I make a new discussion?

Question: The topic "Prover and sequencer market dynamics" what does it mean? Does it mean improving the sequencer like the concept of shared sequencing etc?  Can you please throw some light on the topic?

**Reply by dutterbutter** [reply]
Definitely create a new discussion with the relevant details! Thanks for posting 🚀 🙏

**Comment by andybrierton** [comment]
In zkSync, Hyperchains use a Proof-of-Stake (PoS) consensus mechanism. PoS is utilized to secure the network and reach agreement on the validity of transactions. Consensus is necessary in Hyperchains to ensure the integrity and security of the layer 2 (L2) scaling solution.

While L2 security is indeed reliant on L1, consensus mechanisms like PoS add an additional layer of security and decentralization to the network. It helps prevent attacks and ensures that transactions are processed correctly.

The use of PoS in zkPorter, another component of zkSync, is likely due to its efficiency in achieving consensus while minimizing energy consumption compared to Proof-of-Work (PoW) mechanisms.

In a multi-rollup ecosystem like zkSync, having consensus stake helps maintain the security and integrity of each rollup chain. It ensures that validators have a stake in the network's operation and are incentivized to behave honestly.

Overall, consensus mechanisms play a crucial role in securing and maintaining the reliability of layer 2 scaling solutions like zkSync.

**Comment by backvomeph** [comment]
zkSync hyperchains use a variant of Proof-of-Stake (PoS) consensus mechanism. Consensus is essential for maintaining network integrity and ensuring transaction validity, even in hyperchains. zkPorter, another zkSync component, utilizes PoS for similar reasons, providing security and decentralization. While L2 security relies on L1, consensus mechanisms like PoS help secure the underlying infrastructure and maintain network trust.

**Comment by kennycobbs** [comment]
ZkSync's zkPorter uses a Proof of Stake (PoS) consensus mechanism to secure its hyperchains. Consensus is essential for zkPorter to ensure the validity and finality of transactions within the layer 2 solution. While L2 security is primarily dependent on L1, consensus mechanisms like PoS help maintain the integrity and efficiency of transactions within the layer 2 ecosystem.

**Comment by samui68400** [comment]
**To make things clearer, I'd like to make a comparison with systems already in place.**


If I've understood correctly, hyperchains can be configured as required, with a mix of zk rollups and zk porter.

Can we talk about modularity like celestia, which offsets the availability of off-chain data?

Are hyperchains like rollapps in the dymension ecosystem? In this case Zk Stacks could be considered as cosmos SDK that makes it easy to create hyperchains?

**Reply by bxpana** [reply]
Let's go ahead and open up a new Discussion in the ZK Stack Category and we can answer there.

**Comment by jennifercerso** [comment]
zkSync hyperchains use a consensus mechanism called Proof of Stake (PoS) to validate transactions. While consensus may not be needed for Layer 2 (L2) solutions, it is still important for ensuring the security and integrity of the network. zkPorter, a specific L2 solution within zkSync, uses PoS because it is a proven and secure consensus mechanism. Additionally, while L2 security is dependent on L1, having a consensus mechanism in place on L2 can provide an additional layer of security and protection against potential attacks.

**Comment by andrewtotta** [comment]
The user is seeking clarification on the consensus mechanism used for zkSync hyperchains and whether consensus is needed for L2s. They are also curious about why zkPorter uses POS. The user has not encountered any specific issues or errors and has not provided any relevant code. They have not made any recent changes to the contract and there are no external libraries or contracts being used. The user is seeking information and has not provided any additional details or taken any steps to resolve the issue.
